File: src/backend_dotnet/HRMApi/Models/Participation.cs
using System;
using System.Collections.Generic;
using System.Text.Json;

namespace HRMApi.Models;

public partial class Participation
{
    public int Id { get; set; }

    public int EmployeeId { get; set; }

    public int ActivityId { get; set; }

    public DateTime? RegisterDate { get; set; }

    public DateTime? CancelDate { get; set; }

    public string? Status { get; set; }

    public JsonDocument? Result { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual Activity Activity { get; set; } = null!;

    public virtual Employee Employee { get; set; } = null!;
}

File: src/backend_java/src/main/java/com/group07/human_resource_management/entity/Participation.java
package com.group07.human_resource_management.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(
        name = "participation",
        uniqueConstraints = @UniqueConstraint(columnNames = {"employee_id", "activity_id"})
)
@Getter @Setter
@Builder @NoArgsConstructor @AllArgsConstructor
public class Participation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // FK
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    // FK
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "activity_id", nullable = false)
    private Activity activity;

    @Column(name = "register_date")
    private LocalDateTime registerDate;

    @Column(name = "cancel_date")
    private LocalDateTime cancelDate;

    private String status;

    private String result;

    @Column(name = "created_at")
    private LocalDateTime createdAt;
}
